db/migrate/20200326164755_change_oidc_audience_setting_type.rb

```diff
@@ -0,0 +1,19 @@
+class ChangeOidcAudienceSettingType < ActiveRecord::Migration[5.2]
+  def up
+    setting = Setting.find_by :name => 'oidc_audience'
+    return unless setting
+    setting.value = [setting.value] if setting.value.is_a?(String)
+    setting.settings_type = 'array'
+    setting.default = []
+    setting.save
+  end
+
+  def down
+    setting = Setting.find_by :name => 'oidc_audience'
+    return unless setting
+    setting.value = setting.value.join() if setting.value.is_a?(Array)
```

Comment:
this will lead to `""` if there is an empty array instead of `nil`, and will lead to an invalid value if there is more than one setting.
While rollback is generally not expected, I would prefer to keep the down migration symmetrical by going with `setting.value.first`, even at the risk of losing information.

Reply:
@tbrisker done. I am ok with the risk of loosing information too, users can always realise the loss through the logs while validation of the token.